commands/src/main/java/net/apartium/cocoabeans/commands/parsers/MapBasedParser.java

```diff
@@ -0,0 +1,90 @@
+package net.apartium.cocoabeans.commands.parsers;
+
+import net.apartium.cocoabeans.commands.CommandProcessingContext;
+
+import java.util.*;
+import java.util.function.Supplier;
+
+public abstract class MapBasedParser<T> extends ArgumentParser<T> {
+
+    public abstract Map<String, T> getMap();
+
+    public MapBasedParser(String keyword, Class<T> clazz, int priority) {
+        super(keyword, clazz, priority);
+    }
+
+    @Override
+    public Optional<ParseResult<T>> parse(CommandProcessingContext commandProcessingContext) {
+        List<String> args = commandProcessingContext.args();
+        int index = commandProcessingContext.index();
+
+        Map<String, T> map = getMap();
+        String s = "";
+        for (int i = index; i < args.size(); i++) {
+            if (i != index)
+                s += " ";
+            s += args.get(index);
```

Comment:
This is a very weird behavior that can lead to a lot of confusion, are you sure this is needed?

Reply:
yeah if you have

Map<String, ?> map = Map.of("number ten", 10, "boolean false", false);